src/target/ext/host_io.rs

```diff
@@ -358,15 +335,8 @@ define_ext!(HostIoUnlinkOps, HostIoUnlink);
 pub trait HostIoReadlink: HostIo {
     /// Read value of symbolic link `filename` on the target.
     ///
-    /// The data read _must_ be sent by calling [`HostIoOutput::write`], which
-    /// will consume the `output` object and return a [`HostIoToken`]. This
-    /// token ensures that the implementer of this method calls
-    /// [`HostIoOutput::write`].
-    fn readlink<'a>(
-        &mut self,
-        filename: &[u8],
-        output: HostIoOutput<'a>,
-    ) -> HostIoResult<HostIoToken<'a>, Self>;
+    /// Return the number of bytes written into `buf`.
```

Comment:
update the docs with a note regarding what to do if the buffer doesn't have enough space.

```suggestion
    /// Return the number of bytes written into `buf`.
    ///
    /// Unlike most other Host IO handlers, if the resolved file path exceeds
    /// the length of the provided `buf`, the target should NOT return a
    /// partial response, and MUST return a `Err(HostIoErrno::ENAMETOOLONG)`.
```